src/Package/Impl/ProjectSystem/Commands/SendFileCommandBase.cs

```diff
@@ -32,70 +33,81 @@ internal class SendFileCommandBase {
             _fs = fs;
         }
 
-        protected async Task<bool> SendToRemoteAsync(IEnumerable<string> files, string projectDir, string projectName, string remotePath, CancellationToken cancellationToken) {
-            IVsStatusbar statusBar = _appShell.GetGlobalService<IVsStatusbar>(typeof(SVsStatusbar));
-            return await SendToRemoteWorkerAsync(files, projectDir, projectName, remotePath, statusBar, cancellationToken);
+        protected Task SendToRemoteAsync(IEnumerable<string> files, string projectDir, string projectName, string remotePath) {
+            _appShell.ProgressDialog.Show(async (p, ct) => await SendToRemoteWorkerAsync(files, projectDir, projectName, remotePath, p, ct), Resources.Info_TransferringFiles, 100, 500);
+            return Task.CompletedTask;
         }
 
-        private async Task<bool> SendToRemoteWorkerAsync(IEnumerable<string> files, string projectDir, string projectName, string remotePath, IVsStatusbar statusBar, CancellationToken cancellationToken) {
+        private async Task SendToRemoteWorkerAsync(IEnumerable<string> files, string projectDir, string projectName, string remotePath, IProgress<ProgressDialogData> progress, CancellationToken ct) {
             await TaskUtilities.SwitchToBackgroundThread();
 
-            string currentStatusText;
-            statusBar.GetText(out currentStatusText);
-
             var workflow = _interactiveWorkflowProvider.GetOrCreate();
             var outputWindow = workflow.ActiveWindow.InteractiveWindow;
-            uint cookie = 0;
+
             try {
                 var session = workflow.RSession;
-                statusBar.SetText(Resources.Info_CompressingFiles);
-                statusBar.Progress(ref cookie, 1, "", 0, 0);
-
                 int count = 0;
-                uint total = (uint)files.Count() * 2; // for compressing and sending
+                int total = files.Count();
+
+                progress.Report(new ProgressDialogData(0, Resources.Info_CompressingFiles));
+
+                if (ct.IsCancellationRequested) {
+                    return;
+                }
+
+                List<string> paths = new List<string>();
+                // Compression phase : 1 of 3 phases.
                 string compressedFilePath = string.Empty;
                 await Task.Run(() => {
                     compressedFilePath = _fs.CompressFiles(files, projectDir, new Progress<string>((p) => {
```

Comment:
I have added it. The actual compression code can be a bit annoying. If you cancel and we are in the middle of compressing a massive file. VS is blocked until the file compression is complete.